### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by frontend112**

### CODE REVIEW

**Review by meforoma — CHANGES_REQUESTED** [review]
the code logic is pretty hard to follow to be honest
may i suggest you to approach task using a dateObject (and benefit from `key: value` structure)
Essentially generate a date object (from fromFormat) and later assemble a new date from it.

Hint:
```
const fromSeparator = fromFormat[3];
const dateSplit = date.split(fromSeparator);
const dateObject = {};

  for (let i = 0; i < 3; i++) {
    dateObject[fromFormat[i]] = dateSplit[i];
  }
```
then evaluate YY / YYYY using `.hasOwnProperty` and adjust year format.

And finally assemble a new date.

File: src/formatDate.js [hunk 1]
```diff
@@ -50,7 +50,52 @@
...
+  if (year.length === 2) {
+    // check is it before 2000 or after
+    const thousands = year >= 30 ? 19 : 20;
+
+    //  YY => YYYY
+    newYear = thousands + newYear;
+  }
+
+  return newYear;
 }
```
Comment (ihor-jpeg) [inline src/formatDate.js:99]: write less code
```suggestion
function convertYear(year) {
  if (year.length === 4) {
    // YYYY => YY
    return year.charAt(2) + year.charAt(3);
  }
  
  const thousands = year >= 30 ? 19 : 20;

  //  YY => YYYY
  return thousands + year;
}
```

**Review by ihor-jpeg — APPROVED** [review]

**Review by meforoma — APPROVED** [review]
